pkg/security/security_profile/profile/manager.go

```diff
@@ -361,3 +387,71 @@ func (m *SecurityProfileManager) linkProfile(profile *SecurityProfile, workload
 func (m *SecurityProfileManager) unlinkProfile(profile *SecurityProfile, workload *cgroupModel.CacheEntry) {
 	// TODO: unlink profile <-> container ID in kernel space
 }
+
+func (m *SecurityProfileManager) LookupEventOnProfiles(event *model.Event) {
+	if event.GetEventType() == model.SyscallsEventType {
+		return // syscall matching for anomaly detection is already done kernel side
+	}
+	if event.ContainerContext.ID == "" || len(event.ContainerContext.Tags) < 1 {
+		return
+	}
+	if event.GetEventType() == model.FileOpenEventType && !m.config.SecurityProfileFilesBestEffort {
+		return
+	}
+
+	// if time.Now()-event.ContainerContext.CreatedAt < time.Second*30 {
+	// 	// TODO: put the event in a cache to be pop back after x sec to have a chance to
+	// 	// retrieve a profile for that workload
+	// }
+
+	image := utils.GetTagValue("image_name", event.ContainerContext.Tags)
```

Comment:
wondering if we shouldn't move all this logic of tags == "" and image == "" to the NewWorkloadSelector as we already pass the tags